app/validators/unreserved_validator.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+class UnreservedValidator < ActiveModel::EachValidator
+  def validate_each(record, attribute, value)
+    return if value.nil?
+    record.errors.add(attribute, I18n.t('accounts.reserved_username')) if reserved_username?(value)
+  end
+
+  private
+
+  def reserved_username?(value)
+    return false unless Setting.reserved_usernames
+    Setting.reserved_usernames.include?(value.downcase)
```

Comment:
Could this be one line of `Setting.reserved_usernames.to_a.include?(...)` ... I'm assuming this returns `nil` and not `[]` when it's not set?